coreneuron/nrniv/nrn_checkpoint.cpp

```diff
@@ -428,21 +428,26 @@ static void write_phase2(NrnThread& nt, FileHandlerWrap& fh) {
     for (int i = 0; i < nnetcon; ++i) {
         NetCon& nc = nt.netcons[i];
         Point_process* pnt = nc.target_;
-        assert(pnt);  // nrn_setup.cpp allows type <=0 which generates nullptr target.
-        pnttype[i] = pnt->_type;
+        if (pnt == nullptr) {
+            // nrn_setup.cpp allows type <=0 which generates nullptr target.
+            pnttype[i] = 0;
+            pntindex[i] = -1;
+        } else {
+            pnttype[i] = pnt->_type;
 
 #if 0
-        // todo: this seems most natural, but does not work. Perhaps should look
-        // into how pntindex determined in nrnbbcore_write.cpp and change there.
-        int ix = pnt->_i_instance;
-        if (ml_pinv[pnt->_type]) {
-            ix = ml_pinv[pnt->_type][ix];
-        }
+            // todo: this seems most natural, but does not work. Perhaps should look
+            // into how pntindex determined in nrnbbcore_write.cpp and change there.
+            int ix = pnt->_i_instance;
+            if (ml_pinv[pnt->_type]) {
+                ix = ml_pinv[pnt->_type][ix];
+            }
```

Comment:
This code is in an `#if 0` should we not just drop it? If it's something that doesn't work, but still useful to keep maybe just make those 4 lines comments and remove the `#if 0`. @pramodk ?
In general we should rid the code of these ugly `#if 0` preproc branches.